enterprise/ha/src/main/java/org/neo4j/kernel/ha/SlaveUpdatePuller.java

```diff
@@ -177,25 +206,30 @@ public void run()
     }
 
     @Override
-    public void init() throws Throwable
+    public synchronized void init() throws Throwable
     {
-        // TODO Don't do this. This is just to satisfy LockSupport park/unpark
-        // And we cannot have this class extend Thread since there's a naming clash with Lifecycle for stop()
-        me = new Thread( this, UPDATE_PULLER_THREAD_PREFIX + instanceId );
-        me.start();
+        if ( shutdownLatch != null )
+        {
+            return; // This SlaveUpdatePuller has already been initialised
+        }
+
+        shutdownLatch = new BinaryLatch();
+        jobScheduler.schedule( JobScheduler.Group.pullUpdates, this );
```

Comment:
The previous design used a dedicated thread. I decided to stick with it, because the thread is named, which I think might be a good thing for debugging.